bin/validate-helm

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+cd "$(dirname "$0")"
+
+source ./test-workflow/utils.sh
+source ../helm/common/utils.sh
+
+helm_tests_return_code=0
+
+tests=(happy incorrect_url incorrect_cert incorrect_auth_id)
+
+print_usage() {
+  echo "Usage:"
+  echo "    This script will validate the helm chart tests"
+  echo ""
+  echo "Syntax:"
+  echo "    $0  [Options]"
+  echo "    Options:"
+  echo "    -h            Show help"
+  echo "    -c            Create the Kind cluster"
+  echo "    -d            Delete the Kind cluster"
+  echo "    -t            run one test"
+}
+
+
+function pushd() {
+    command pushd "$@" > /dev/null
+}
+
+function popd() {
+    command popd "$@" > /dev/null
+}
+
+function create_cluster() {
+# Create kind cluster
+    ./0_prep_conjur_in_kind.sh
+    . ./1_prep_env.sh
+    ./2_admin_load_conjur_policies.sh
+    ./3_admin_init_conjur_cert_authority.sh
+}
+
+function delete_cluster() {
+
+    kind delete cluster
+}
+
+readonly APPLIANCE_URL="https://conjur-oss.conjur-oss.svc.cluster.local"
+
+function git_clone() {
+    rm -rf conjur-authn-k8s-client
+    git clone https://github.com/cyberark/conjur-authn-k8s-client.git
+    pushd conjur-authn-k8s-client/helm/conjur-config-cluster-prep
+    ./bin/get-conjur-cert.sh -u $APPLIANCE_URL -i -v -s
+}
+
+function create_cluster_helm_chart() {
+  test_name=$1
+  auth_enable=false
+  appl_url=$APPLIANCE_URL
+  cert_path="files/conjur-cert.pem"
+  auth_id="my-authenticator-id"
+
+  announce "$test_name test"
+
+  if [ "$test_name" = "incorrect_url" ]
+  then
+    appl_url="https://conjer-oss.conjer-oss.svc.cluster.local"
+  fi
+
+  if [ "$test_name" = "incorrect_cert" ]
+  then
+    auth_enable=true
+    cert_path="tests/test-cert.pem"
+  fi
+
+  if [ "$test_name" = "incorrect_auth_id" ]
+  then
+    auth_enable=true
+    auth_id="your-authenticator-id"
+  fi
+
+  helm upgrade --install "$helm_release" . -n conjur-oss --wait \
+               --set test.authentication.enable="$auth_enable" \
+               --set conjur.applianceUrl="$appl_url" \
+               --set conjur.certificateFilePath="$cert_path" \
+               --set authnK8s.authenticatorID="$auth_id"
+
+  if [ $? -eq 0 ]
+  then
+      announce "Helm $helm_release installed successfully"
+  else
+       banner "$RED" "Helm install Failed"
+  fi
+  helm get values "$helm_release"
+}
+
+function test_helm_chart() {
+
+  release=$1
+  test_name=$2
+  expected=1
```

Comment:
Can we be certain that `helm test ...` will always return a 1 upon failure, or just that it will be non-zero?
Might take a few more conditionals, but would be safer to check for zero vs. non-zero?

Reply:
good idea.